lib/rpc/handler/FallbackHandler.ts

```diff
@@ -0,0 +1,165 @@
+import Logger from 'bunyan'
+import { ChainId } from '@uniswap/sdk-core'
+import { getProviderId } from '../utils'
+import { ProviderHealthStateRepository } from '../ProviderHealthStateRepository'
+import { ProviderHealthStateDynamoDbRepository } from '../ProviderHealthStateDynamoDbRepository'
+import { ProviderHealthiness, ProviderHealthState } from '../ProviderHealthState'
+import { MetricLoggerUnit } from '@uniswap/smart-order-router'
+import { metricScope, MetricsLogger } from 'aws-embedded-metrics'
+import { APIGatewayProxyResult } from 'aws-lambda'
+import { AWSMetricsLogger } from '../../handlers/router-entities/aws-metrics-logger'
+
+export interface AlarmEvent {
+  alarmName: string
+  state: string
+  previousState: string // Only for logging purpose. Not used in any logic.
+  providerId: string
+  reason: string // Only for logging purpose. Not used in any logic
+}
+
+export class HealthinessUpdate {
+  constructor(public oldHealthiness: ProviderHealthiness, public newHealthiness: ProviderHealthiness) {}
```

Comment:
Why does the constructor need to have both passed in? My undersatnding is that the old change should be part of the constructor and the new state should be part of isChanged?

Reply:
Refactored this part of code. The readability should be much improved by now.